ast.c

```diff
@@ -622,6 +633,23 @@ static void ast_fill_children_ht(HashTable *ht, zend_ast *ast, ast_state_info_t
 			}
 		}
 
+#if PHP_VERSION_ID >= 80000
+		if (child == NULL) {
+			if (i == 4) {
+				if (ast_kind == ZEND_AST_PARAM || ast_kind == ZEND_AST_METHOD || ast_kind == ZEND_AST_FUNC_DECL || ast_kind == ZEND_AST_CLOSURE || ast_kind == ZEND_AST_ARROW_FUNC) {
+					continue;
+				}
+			} else if (i == 3) {
+				if (ast_kind == ZEND_AST_PARAM || ast_kind == ZEND_AST_CLASS) {
+					continue;
+				}
+			} else if (i == 2) {
+				if (ast_kind == ZEND_AST_PROP_GROUP) {
+					continue;
+				}
+			}
+		}
+#endif
```

Comment:
Am I understanding right that this code prevents inclusion of `attributes` if they aren't set? I don't think that's a good idea, we should always provide the child, even if it's null.

Reply:
> Am I understanding right that this code prevents inclusion of attributes if they aren't set? I don't think that's a good idea, we should always provide the child, even if it's null.

Yes. A viable alternative would be excluding them in version 70 in below even if they are set, and setting them to null/the attribute list in version 80 in all php versions.

Reply:
That sounds reasonable to me.